overlord/configstate/configcore/utils.go

```diff
@@ -27,7 +27,7 @@ import (
 )
 
 // first match is if it is comment, second is key, third value
-var rx = regexp.MustCompile(`^[ \t]*(#?)[ \t#]*([a-z_]+)=(.*)$`)
+var rx = regexp.MustCompile(`^[ \t]*(#?)[ \t#]*([a-zA-Z0-9_]+)=(.*)$`)
```

Comment:
This is also matching uppercase. Is the file able to handle other casing, and more importantly is it working correctly? What happens if the key in the file is uppercase but our configuration is not, considering that our configuration keys in snapd can only be lowercase?